### DISCUSSION TITLE
Does FreeTDS ODBC driver should be maintained continually?

### DISCUSSION THREAD

**Discussion by taozuhong** [FreeTDS/freetds#593] [General]
[Microsoft SQL Server ODBC driver](https://learn.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16) has been landed on Linux and macOS, so I have a question:

Does FreeTDS ODBC driver should be maintained continually?

Maybe FreeTDS should drop the ODBC support, focus on the light-weight library for SQL Server(libtds), and wrapper its api as **ct-lib** and **db-lib**.

**Comment by freddy77** [comment]
ODBC library is the only library getting additional improvement, lot of usages of ctlib/dblib came from porting. Adding extensions to ctlib/dblib is doable and done but possibly limiting. Also the world is not all Windows, Linux and MacOS and as far as I know Microsoft driver is not open source.

**Comment by taozuhong** [comment]
Ah, if missing ODBC driver, we have JDBC and ADBC, or access via `libtds` directly, so focus on `libtds` and its porting library(ctlib/dblib) or just `libtds` only, it seems that is the right choice.

Maybe FreeTDS could split into three projects: libtds / libtdssql(ctlib/dblib) / libtdsodbc, `libtds` is the core engine, focus on making it powerful, portable and stable, it will release more power.

We have limited energy, so focus on what is most valuable（`libtds`).

**Comment by mmcnabb-vms** [comment]
I am using FreeTDS & unixODBC, on a platform which is not Linux, MacOS or Windows

**Comment by paraddise** [comment]
Freetds now can connect to mssql servers with EPA enabled and required, because the official ODBC driver from Microsoft can't do it on linux and macos.
